tree/dataframe/src/RCsvDS.cxx

```diff
@@ -195,6 +195,8 @@ void RCsvDS::InferColTypes(std::vector<std::string> &columns)
 
 void RCsvDS::InferType(const std::string &col, unsigned int idxCol)
 {
+   R__LOCKGUARD(gROOTMutex);
+
```

Comment:
why does this need locking? I see the `TRegexp`s are static, but do they _have_ to be? (@etejedor might have an opinion as RCsvDS author)

Reply:
We can just declare the statics as `static const`, missed that before.